### ISSUE TITLE
Pillar render error: Failed to load ext_pillar s3: global name '__utils__' is not defined

### ISSUE THREAD

**Issue by icflournoy** [saltstack/salt#37894]
### Description of Issue/Question

I'm attempting to use the S3 pillar source with a simple configuration, but am encountering the following critical output:

`[CRITICAL] Pillar render error: Failed to load ext_pillar s3: global name '__utils__' is not defined`

### Setup

**IAM Instance role - relevant S3 access policy**
```
{
  "Version": "2012-10-17",
  "Statement": [
    {
      "Sid": "1",
      "Effect": "Allow",
      "Action": [
        "s3:ListBucket",
        "s3:GetObject"
      ],
      "Resource": [
        "arn:aws:s3:::company-salt-pillars/*",
        "arn:aws:s3:::company-salt-pillars"
      ]
    }
  ]
}
```

**salt-master relevant config**
```
ext_pillar:
  - s3:
      bucket: company-salt-pillars
```

**S3 bucket contents**
Disclaimer: I haven't used pillars before so I'm not sure if I'm representing the pillar structure in the S3 bucket properly. However, it doesn't seem like I'm even getting to the step where the S3 pillar accesses the bucket.
```
icflournoy@computer: [~]  12:21:28
$ aws s3 ls --recursive company-salt-pillars
2016-11-24 23:10:50          0 base/
2016-11-24 23:10:59         14 base/test_pillar
2016-11-24 23:16:08         14 base/test_pillar.sls
2016-11-24 23:16:09         31 base/top.sls
```

### Steps to Reproduce Issue
 1. I start both the salt-master and salt-minion in debug mode.
 2. I then run `salt 'salt-master*' saltutil.refresh_pillar` and look at the master's log.

**salt-master debug log**
```
[DEBUG   ] LazyLoaded s3.ext_pillar
[DEBUG   ] S3 bucket cache file /var/cache/salt/master/pillar_s3fs/company-salt-pillars--files.cache is expired, mtime_diff=-1480093825.35s, expiration=30s
[DEBUG   ] Refreshing S3 buckets pillar cache file
[DEBUG   ] Single environment per bucket mode
[DEBUG   ] LazyLoaded s3.query
[CRITICAL] Pillar render error: Failed to load ext_pillar s3: global name '__utils__' is not defined
```

Alternatively if I run `salt 'salt-master*' pillar.items` I am able to see the same error message:
```
root@salt-master-01a:~# salt 'salt-master*' pillar.items
salt-master-01a.us-west-2.company.com:
    ----------
    _errors:
        - Failed to load ext_pillar s3: global name '__utils__' is not defined
    local_pillar:
        True
```

### Versions Report
```
root@salt-master-01a:~# salt --versions-report
Salt Version:
           Salt: 2016.3.4
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: 3.5.0
       dateutil: 2.4.2
          gitdb: 0.6.4
      gitpython: 1.0.1
          ioflo: Not Installed
         Jinja2: 2.8
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: 1.0.3
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.7.12 (default, Nov 19 2016, 06:48:10)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.2.0
           RAET: Not Installed
          smmap: 0.9.0
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.4
 
System Versions:
           dist: Ubuntu 16.04 xenial
        machine: x86_64
        release: 4.4.0-36-generic
         system: Linux
        version: Ubuntu 16.04 xenial
```

**Comment by w32-blaster** [comment]
I have the same problem using different commands:
- `salt '*salt*' s3.get s3-bucket`
- Having an s3 source in a file.managed state. 

The same salt version, same OS version

**Comment by icflournoy** [comment]
@w32-blaster I can corroborate the `s3.get` behavior as well.

```
root@salt-master-01a:~# salt 'salt-master*' s3.get company-salt-pillars
salt-master-01a.us-west-2.company.com:
    The minion function caused an exception: Traceback (most recent call last):
      File "/usr/lib/python2.7/dist-packages/salt/minion.py", line 1363, in _thread_return
        return_data = executor.execute()
      File "/usr/lib/python2.7/dist-packages/salt/executors/direct_call.py", line 28, in execute
        return self.func(*self.args, **self.kwargs)
      File "/usr/lib/python2.7/dist-packages/salt/modules/s3.py", line 182, in get
        role_arn=role_arn)
      File "/usr/lib/python2.7/dist-packages/salt/utils/s3.py", line 100, in query
        location = __utils__['aws.get_location']()
    NameError: global name '__utils__' is not defined
```

**Comment by icflournoy** [comment]
I was able to make both `s3.get` and my original `pillar.items` work by specifying the location to perform the query from. (I'm running this on an instance in the `us-west-2` region so that's what I've specified.)

This is using v2016.3.4.

```
root@salt-master-01a:/usr/lib/python2.7/dist-packages/salt# salt-call --local s3.get company-salt-pillars base/top.sls location=us-west-2 return_bin=True
[WARNING ] /usr/lib/python2.7/dist-packages/urllib3/connectionpool.py:794: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.org/en/latest/security.html
  InsecureRequestWarning)

local:
    base:
      '*':
        - test_pillar
```


This works because once you explicitly define a location it means this code never executes https://github.com/saltstack/salt/blob/v2016.3.4/salt/utils/s3.py#L100

This issue is fixed in #37428 - and the fix works for me since I've checked out the latest version and tested without explicitly specifying a location.

```
(salt-virtualenv) root@salt-master-01a:/opt/salt-virtualenv# salt --version
salt 2016.11.0-415-g843ceac (Carbon)
(salt-virtualenv) root@salt-master-01a:/opt/salt-virtualenv# salt-call -c etc/salt/ --local s3.get company-salt-pillars base/top.sls return_bin=True
# snipped warnings

local:
    base:
      '*':
        - test_pillar
```


Now time to actually make the pillar correct...

**Comment by icflournoy** [comment]
I spoke too soon. While #37428  fixed the `s3.get` command it didn't entirely fix the S3 pillar.

When refreshing pillars I encountered a very similar issue to the same one I initially submitted except this time `__opts__`  isn't defined. That's fixed in #37629
